app/store/migrations/075.ts

```diff
@@ -0,0 +1,422 @@
+  type CaipAccountId,
+  type Json,
+  type Hex,
+  type NonEmptyArray,
+} from '@metamask/utils';
+import type {
+  Caveat,
+  PermissionConstraint,
+  ValidPermission,
+} from '@metamask/permission-controller';
+
+// In-lined from @metamask/multichain
+const Caip25CaveatType = 'authorizedScopes';
+const Caip25EndowmentPermissionName = 'endowment:caip25';
+
+interface InternalScopeObject {
+  accounts: CaipAccountId[];
+}
+
+type InternalScopesObject = Record<CaipChainId, InternalScopeObject>;
+
+interface Caip25CaveatValue {
+  requiredScopes: InternalScopesObject;
+  optionalScopes: InternalScopesObject;
+  sessionProperties?: Record<string, Json>;
+  isMultichainOrigin: boolean;
+}
+
+// Locally defined types
+//@ts-expect-error Caip25CaveatValue is a valid Caveat Value
+type Caip25Caveat = Caveat<typeof Caip25CaveatType, Caip25CaveatValue>;
+type Caip25Permission = ValidPermission<
+  typeof Caip25EndowmentPermissionName,
+  //@ts-expect-error this is a valid permission
+  Caip25Caveat
+>;
+
+const PermissionNames = {
+  eth_accounts: 'eth_accounts',
+  permittedChains: 'endowment:permitted-chains',
+} as const;
+
+// a map of the networks built into the extension at the time of this migration to their chain IDs
+// copied from shared/constants/network.ts (https://github.com/MetaMask/metamask-extension/blob/5b5c04a16fb7937a6e9d59b1debe4713978ef39d/shared/constants/network.ts#L535)
+const BUILT_IN_NETWORKS: ReadonlyMap<string, Hex> = new Map([
+  ['sepolia', '0xaa36a7'],
+  ['mainnet', '0x1'],
+  ['linea-sepolia', '0xe705'],
+  ['linea-mainnet', '0xe708'],
+]);
+
+const snapsPrefixes = ['npm:', 'local:'] as const;
+
+function isPermissionConstraint(obj: unknown): obj is PermissionConstraint {
+  return (
+    isObject(obj) &&
+    obj !== null &&
+    hasProperty(obj, 'caveats') &&
+    Array.isArray(obj.caveats) &&
+    obj.caveats.length > 0 &&
+    hasProperty(obj, 'date') &&
+    typeof obj.date === 'number' &&
+    hasProperty(obj, 'id') &&
+    typeof obj.id === 'string' &&
+    hasProperty(obj, 'invoker') &&
+    typeof obj.invoker === 'string' &&
+    hasProperty(obj, 'parentCapability') &&
+    typeof obj.parentCapability === 'string'
+  );
+}
+
+function isNonEmptyArrayOfStrings(obj: unknown): obj is NonEmptyArray<string> {
+  return (
+    Array.isArray(obj) &&
+    obj.length > 0 &&
+    obj.every((item) => typeof item === 'string')
+  );
+}
+
+/**
+ * Migration to update the legacy `eth_accounts` and `endowment:permitted-chains` state to `endowment:caip25` state.
+ * @param state - The current MetaMask mobile state.
+ * @returns Migrated Redux state.
+ */
+export default function migrate(oldState: unknown) {
+  const version = 75;
+
+  // Ensure the state is valid for migration
+  if (!ensureValidState(oldState, version)) {
+    return oldState;
+  }
+
+  const newState = cloneDeep(oldState);
+  const { backgroundState } = newState.engine;
+
+  if (!hasProperty(backgroundState, 'PermissionController')) {
+    return oldState;
```

Comment:
This seems something that would bring problems, or the PermissionController since it doesn't exist, will create it's state once the app initializes? If not, should we capture the exception to understand what could lead to a non existing PermissionController?
```suggestion
  new Error(
        `Migration ${version}: PermissionController does not exist`,
      ),
    );
    return oldState;
```

Reply:
This scenario may occur on fresh app install where no state exists yet. Should not be a problem as I understand. We are doing the same in extension migrations as well

Reply:
Hmmm if migrations are running in fresh app install, something is not working properly I believe

Reply:
I'm not entirely sure if mobile starts at a migration version of 0 or the most recent migration. I believe it would start at 0, but i could be wrong

Reply:
done here https://github.com/MetaMask/metamask-mobile/pull/13970/commits/6b351d45b0d736c8a011b0763be110b02cf24b88